File: CodeWars/8K-Reversed_Strings.cs
/* 
Level: 8 kyu
Link: https://www.codewars.com/kata/5168bb5dfe9a00b126000018/train/csharp
*/

using System;

public static class Kata
{
  public static string Solution(string str) 
  {
    char[] charArray = str.ToCharArray();
    Array.Reverse(charArray);
    
    return new String (charArray);
  }
}

// Solution was googled

/* Top solutions analysis */
using System;
using System.Linq;

public static class Kata
{
  public static string Solution(string str) 
  {
     return new string(str.ToArray().Reverse().ToArray());
  }
}

File: CodeWars/Javascript/8K-Reversed_Strings.js
/* 
Level: 8 kyuu
Link: https://www.codewars.com/kata/5168bb5dfe9a00b126000018
*/

function solution(str){
  
    let reversedString = '';
    
    for(let i = str.length - 1; i >= 0; i--) {
      reversedString += str[i];
    }
    
    return reversedString;
    
  }

/* Top solutions analysis */

function solution(str){
    return str.split('').reverse().join('');  
  }

// split() method is splitting the string into an array of letters
// reverse() method is reversing the array
// join () method is used to concatenate all the elements of the array